lib/msal-common/src/request/RequestValidator.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ScopeSet } from "./ScopeSet";
+import { StringUtils } from "./../utils/StringUtils";
+import { ClientConfigurationError } from "./../error/ClientConfigurationError";
+import { PromptValue, CodeChallengeMethodValues } from "./../utils/Constants";
+
+/**
+ * Validates server consumable params from the "request" objects
+ */
+export class RequestValidator {
```

Comment:
If we have classes for the request objects we should put the validation in those classes, the only reason to have a separate class for validation is if we are using types.

Reply:
nope. The intent is to have all/any request objects be able to call each validator as it applies. Adds flexibility for various requests that will come as we keep adding support for different flows.